app/models/components/course/duplication_ability_component.rb

```diff
@@ -24,8 +26,20 @@ def disallow_superusers_duplicate_via_frontend
     cannot :duplicate_from, Course
   end
 
-  def disallow_users_to_duplicate_cross_instances
-    cannot :duplicate_to_other_instances, Instance
+  def allow_administrator_to_duplicate_cross_instances
+    can :duplicate_across_instances, Instance
+  end
+
+  def allow_instance_admin_to_duplicate_cross_instances
+    can :duplicate_across_instances, Instance do |instance|
+      instance.instance_users.administrator.exists?(user_id: user.id)
```

Comment:
Can you check that there is no performance issue here? Why use `exists` over `present?` or `any?`?

Reply:
the reason exists is used here is because
1. we only care whether such administrator exists (we don't need to care about how many of them are)
2. exists? will stop the iteration after one such object is found, and immediately return true (the SQL code generated here has LIMIT 1 in there, ensuring that once one such object is found then return true immediately)

I was also thinking to split the ability into `to` and `from`, however this will only add unnecessary complexity and redundancy to the code, while this ability is specific only for duplicating instance, and it's always clear that to duplicate across instance, the ability needs to be there for source and destination.